File: frontend/src/components/SuperChat/super_chat.ts
import type {User} from "../types";


export interface SuperChat{
    user:User
    content:string
    room_id:number
    message_id:string
    timestamp:number
    end_time:number
    price:number
}

export function computeSuperChatBackground(price :number):string{
    switch (price){
        case 30:
            return "#ff0000"
        case 50:
            return "#F8D766"
    }
    return ""
}

File: internal/model/super_chat.go
package model

type SuperChat struct {
	User      User   `json:"user"`
	Content   string `json:"content"`
	RoomID    int    `json:"room_id"`
	MessageID string `json:"message_id"`
	Timestamp int    `json:"timestamp"`
	EndTime   int    `json:"end_time"`
	Price     int    `json:"price"`
}
